core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/SpanLabelView.java

```diff
@@ -62,7 +62,17 @@ public SpanLabelView(String viewName, String viewGenerator, TextAnnotation text,
     public void addConstituent(Constituent constituent) {
         super.addConstituent(constituent);
 
-        Collections.sort(this.constituents, TextAnnotationUtilities.constituentStartComparator);
```

Comment:
is it worth simply replacing the map with a TreeMap, as you suggest? if not now, please open an issue.

Reply:
This is an inherited data structure. There is already an issue open for this. I will revise the existing issue.